### PR TITLE
lncli: replace deprecated fork for serializing protobuf as JSON with new one

### PR DESCRIPTION
**Pull request by guggero**
Depends on https://github.com/lightninglabs/protobuf-go-hex-display/pull/1.

Replaces the [old fork]() of the `github.com/golang/protobuf/jsonpb` module with a new one based on the `google.golang.org/protobuf/encoding/protojson` one instead.

### CODE REVIEW

**Review by bhandras — APPROVED** [review]
tACK, changes LGTM 🥇

**Review by bitromortac — APPROVED** [review]
Very nice cleanup PR :broom: LGTM. Tested lncli marshal/unmarshal (sendtoroute) and a REST query.

File: cmd/lncli/cmd_payments.go [hunk 1]
```diff
@@ -952,7 +951,7 @@ func sendToRoute(ctx *cli.Context) error {
 	// format.
 	var route *lnrpc.Route
 	routes := &lnrpc.QueryRoutesResponse{}
-	err = jsonpb.UnmarshalString(jsonRoutes, routes)
+	err = lnrpc.ProtoJSONUnmarshalOpts.Unmarshal([]byte(jsonRoutes), routes)
```
[the remark this answers is not in the document]
Comment (guggero) [reply cmd/lncli/cmd_payments.go:954]: I don't think so. I guess the only reason the `Unmarshal` function takes a byte slice is to make it easy to pass in bytes read from a file directly. But there should be no issue with the direct cast here.

File: go.mod [hunk 2]
```diff
@@ -56,7 +56,7 @@ require (
 	golang.org/x/term v0.6.0
 	golang.org/x/time v0.0.0-20210220033141-f8bda1e9f3ba
 	google.golang.org/grpc v1.41.0
-	google.golang.org/protobuf v1.27.1
+	google.golang.org/protobuf v1.30.0
```
[the remark this answers is not in the document]
Comment (guggero) [reply go.mod:59]: Good catch, fixed.

**Pull request merged**